### PR TITLE
e2e case of operating tidb-cluster without helm

### LINKED ISSUE #1378: e2e case of deploy TiDB cluster without helm
#1121

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

close #1378 

### Check List 

Tests 

 - Unit test
 - E2E test

### Does this PR introduce a user-facing change?:
 
 ```release-note
e2e case of operating tidb-cluster without helm
 ```

A fix of TiKVConfig schema is also included, please check https://github.com/pingcap/tidb-operator/pull/1396/commits/97a075a361b6e1ae83c6ad1128c5e92275ab9b90 for non-codegen change

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/tikv_config.go [hunk 1]
```diff
@@ -26,56 +26,7 @@ type TiKVConfig struct {
 	LogRotationTimespan string `json:"log-rotation-timespan,omitempty" toml:"log-rotation-timespan,omitempty"`
 	// +optional
 	PanicWhenUnexpectedKeyOrData *bool `json:"panic-when-unexpected-key-or-data,omitempty" toml:"panic-when-unexpected-key-or-data,omitempty"`
-	// +optional
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/tikv_config.go]: why these fields are removed?

Comment (aylei) [reply pkg/apis/pingcap/v1alpha1/tikv_config.go]: They are placed to a wrong place (should be under the `[server]` section)

File: pkg/apis/pingcap/v1alpha1/tikv_config.go [hunk 2]
```diff
@@ -456,48 +407,57 @@ type TiKVBlockCacheConfig struct {
...
-	StatsConcurrency *int64 `json:"stats-concurrency,omitempty" toml:"stats-concurrency,omitempty"`
+	StatsConcurrency *uint `json:"stats-concurrency,omitempty" toml:"stats-concurrency,omitempty"`
 	// +optional
-	HeavyLoadThreshold *int64 `json:"heavy-load-threshold,omitempty" toml:"heavy-load-threshold,omitempty"`
+	HeavyLoadThreshold *uint `json:"heavy-load-threshold,omitempty" toml:"heavy-load-threshold,omitempty"`
 	// +optional
 	HeavyLoadWaitDuration string `json:"heavy-load-wait-duration,omitempty" toml:"heavy-load-wait-duration,omitempty"`
 	// +optional
 	Labels map[string]string `json:"labels,omitempty" toml:"labels,omitempty"`
+	// +optional
```
Comment (tennix) [inline pkg/apis/pingcap/v1alpha1/tikv_config.go]: Remove this.

File: pkg/controller/controller_utils.go [hunk 3]
```diff
@@ -444,3 +447,21 @@ func InferObjectKind(obj runtime.Object) (schema.GroupVersionKind, error) {
 	}
 	return gvks[0], nil
 }
+
+// GuaranteedUpdate will retry the updateFunc to mutate the object util success, updateFunc is expected to
```
Comment (tennix) [inline pkg/controller/controller_utils.go]:
```suggestion
// GuaranteedUpdate will retry the updateFunc to mutate the object until success, updateFunc is expected to
```

File: pkg/controller/controller_utils.go [hunk 4]
```diff
@@ -444,3 +447,21 @@ func InferObjectKind(obj runtime.Object) (schema.GroupVersionKind, error) {
...
+// capture the object reference from the caller context to avoid unnecessary type casting.
+func GuaranteedUpdate(cli client.Client, obj runtime.Object, updateFunc func() error) error {
```
Comment (tennix) [inline pkg/controller/controller_utils.go]: Will this be used in the controller in the future? Seems this is used in test only.

Comment (aylei) [reply pkg/controller/controller_utils.go]: Yes, some get-modify-update logic could be simplified with this method

**Pull request merged**